isis/src/base/objs/StretchBlob/StretchBlob.cpp

```diff
@@ -51,67 +60,52 @@ namespace Isis {
     return m_stretch;
   }
 
-  /**
-   * Read saved Stretch data from a Cube into this object. 
-   *  
-   * This is called by Blob::Read() and is the actual data reading function 
-   * ultimately called when running something like cube->read(stretch);  
-   * 
-   * @param is input stream containing the saved Stretch information
-   */
-  void StretchBlob::ReadData(std::istream &is) {
-    // Set the Stretch Type
-     m_stretch.setType(p_blobPvl["StretchType"][0]);
-     m_stretch.setBandNumber(p_blobPvl["BandNumber"][0].toInt());
-
-     // Read in the Stretch Pairs
-     streampos sbyte = p_startByte - 1;
-     is.seekg(sbyte, std::ios::beg);
-     if (!is.good()) {
-       QString msg = "Error preparing to read data from " + m_stretch.getType() +
-                    " [" + p_blobName + "]";
-       throw IException(IException::Io, msg, _FILEINFO_);
-     }
-
-     char *buf = new char[p_nbytes+1];
-     memset(buf, 0, p_nbytes + 1);
-
-     is.read(buf, p_nbytes);
-
-     // Read buffer data into a QString so we can call Parse()
-     std::string stringFromBuffer(buf);
-     m_stretch.Parse(QString::fromStdString(stringFromBuffer));
-
-     delete [] buf;
-
-     if (!is.good()) {
-       QString msg = "Error reading data from " + p_type + " [" +
-                    p_blobName + "]";
-       throw IException(IException::Io, msg, _FILEINFO_);
-     }
-   }
 
-
-  /**
-   *  Initializes for writing stretch to cube blob
-   */
-  void StretchBlob::WriteInit() {
-    p_nbytes = m_stretch.Text().toStdString().size(); 
+  Isis::Blob *StretchBlob::toBlob() {
+    Isis::Blob *blob = new Blob(m_stretch.getName(), m_stretch.getType());
```

Comment:
The Stretch Blobs are named "CubeStretch" and have type "Stretch", which is different from the name of the specific CubeStretch name, specified by the user and could be anything, like "MyBestStretch2021!" and the type is the specific type of stretch specified in the AdvancedStretchTool (i.e. Linear, Sawtooth, etc)

Reply:
So, the stretch name should be different than the blob name, for example.